daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/processor/TDMLDFDLProcessor.scala

```diff
@@ -141,3 +144,12 @@ trait TDMLUnparseResult extends TDMLResult {
   def isScannable: Boolean
   def encodingName: String
 }
+
+trait TDMLErrorHandler extends ErrorHandler with WithDiagnostics {
+  override def warning(exception: SAXParseException): Unit
+  override def error(exception: SAXParseException): Unit
+  override def fatalError(exception: SAXParseException): Unit
+
+  override def getDiagnostics: Seq[Diagnostic]
+  override def isError: Boolean
+}
```

Comment:
I think the TDML Error handler is completely internal to the DaffodilTDMLDFDLProcessor, right? Note sure you need this trait. Your DaffodilSAXErrorHandler can just extend ErrorHandler without this trait.